cli/azd/test/functional/telemetry_test.go

```diff
@@ -368,10 +374,24 @@ func verifyResource(t *testing.T, attributes []Attribute) {
 
 	require.Contains(t, m, fields.ExecutionEnvironmentKey)
 
+	env := ""
 	if os.Getenv("BUILD_BUILDID") != "" {
-		require.Equal(t, m[fields.ExecutionEnvironmentKey], fields.EnvAzurePipelines)
+		env = fields.EnvAzurePipelines
+		require.Regexp(t, regexp.MustCompile("^"+fields.EnvAzurePipelines), m[fields.ExecutionEnvironmentKey])
 	} else if os.Getenv("GITHUB_RUN_ID") != "" {
-		require.Equal(t, m[fields.ExecutionEnvironmentKey], fields.EnvGitHubActions)
+		env = fields.EnvGitHubActions
+	}
+
+	if env != "" {
+		// basic regex that matches a very simple expression (not the entire grammar):
+		// env followed by an optional (;modifier)
+		require.Regexp(t, regexp.MustCompile("^"+env+"(;\\w)?"), m[fields.ExecutionEnvironmentKey])
```

Comment:
FWIW, sometimes the linter yells at me to use raw strings (via backticks) when writing regular expressions so you don't have to worry about escaping `\`.

Reply:
Will keep this in mind for future changes.